### ISSUE TITLE
Variable kubernetes::containerd_config_source is not honored 

### ISSUE THREAD

**Issue by gaure** [puppetlabs/puppetlabs-kubernetes#559]
## Describe the Bug
Hi 
I am trying to use the variable 
kubernetes::containerd_config_source: puppet:///module/profile/kubernetes/worker/config.toml 
In a worker hiera node.yaml file and it is not working.
The module ignore it a continue using the template.

## Expected Behavior
The /etc/containerd/config.toml use source puppet:///module/profile/kubernetes/worke/config.toml instead of the erb template

## Steps to Reproduce
1. add variable:  "kubernetes::containerd_config_source: puppet:///module/profile/kubernetes/worker/config.toml" to the hiera node.yaml file along with the other kubernetes module variables.
2. Assign to that node a role with a profile that has the kubernetes class, example:
```
  class { 'kubernetes':
    cgroup_driver => 'systemd',
    worker        => true,
}
```
3. Run the puppet agent

 - Version [e.g. 6.21.1]
 - Platform [e.g. Ubuntu 20.04]

## Additional Context
Other than that the module is working perfectly

**Comment by gaure** [comment]
Hi 

I found the problem, and it is related to the documentation.
If you install containerd using the "archive" option the variable containerd_config_source does not have any function.
The variable is only applied if you install containerd with the "package" option.
Apparently the "if" condition that decides is the file content will come from the module template or from your custom sources is does not exists in the  "archive" installation mode logic.

Closing this bug.
